File: s1873.c
#include <stdio.h>

int main(void)
{
  int N, i, j;
  char v[16], s, r;

  scanf("%d", &N);
  for (i = 0; i < N; i++)
  {
    scanf(" %[^\n]s", v);
    for (j = 0; v[j] != ' '; j++)
      ;
    r = v[2];
    s = v[j + 3];
    if (r == 's')
    {
      if (s == 'p' || s == 'g')
      {
        printf("rajesh\n");
      }
      else if (s == 'd' || s == 'o')
      {
        printf("sheldon\n");
      }
      else
      {
        printf("empate\n");
      }
    }
    else if (r == 'p')
    {
      if (s == 'd' || s == 'o')
      {
        printf("rajesh\n");
      }
      else if (s == 's' || s == 'g')
      {
        printf("sheldon\n");
      }
      else
      {
        printf("empate\n");
      }
    }
    else if (r == 'd')
    {
      if (s == 'g' || s == 's')
      {
        printf("rajesh\n");
      }
      else if (s == 'p' || s == 'o')
      {
        printf("sheldon\n");
      }
      else
      {
        printf("empate\n");
      }
    }
    else if (r == 'g')
    {
      if (s == 'p' || s == 'o')
      {
        printf("rajesh\n");
      }
      else if (s == 's' || s == 'd')
      {
        printf("sheldon\n");
      }
      else
      {
        printf("empate\n");
      }
    }
    else
    {
      if (s == 's' || s == 'd')
      {
        printf("rajesh\n");
      }
      else if (s == 'p' || s == 'g')
      {
        printf("sheldon\n");
      }
      else
      {
        printf("empate\n");
      }
    }
  }
  return 0;
}

File: s1873.js
var input = require("fs").readFileSync(__dirname + "/dev/stdin/input.txt", "utf8");
var lines = input.split("\r").join("").split("\n");

let regras = {
  tesoura: ["papel", "lagarto"],
  papel: ["pedra", "spock"],
  pedra: ["lagarto", "tesoura"],
  lagarto: ["spock", "papel"],
  spock: ["tesoura", "pedra"],
};

let N = parseInt(lines.shift());
for (let ct = 0; ct < N; ct++) {
  let line = lines.shift().toLowerCase().split(" ");
  let Rajesh = line[0];
  let Sheldon = line[1];
  if (regras[Rajesh].includes(Sheldon)) {
    console.log("rajesh");
  } else if (regras[Sheldon].includes(Rajesh)) {
    console.log("sheldon");
  } else {
    console.log("empate");
  }
}
